### DISCUSSION TITLE
Access the output of local-embedding network

### DISCUSSION THREAD

**Discussion by luigibonati** [deepmodeling/deepmd-kit#52] [Q&A]
Hi all,

I wanted to ask you whether it is possible to print the output of the local-embedding network, in order to check the inputs of the fitting network. 

If this is not implemented yet, could you please give some directions about the part of the code that needs to be modified in order to get it? 

Thank you, 

Luigi

**Comment by amcadmus** [comment]
Hi Luigi

Please take a look at the code at the devel branch. 

When freeze the model, please add "o_descriptor" to the default node list by the command line argument "-n", e.g.
python -m deepmd freeze -n 
o_energy,o_force,o_virial,o_atom_energy,o_atom_virial,descrpt_attr/rcut,descrpt_attr/ntypes,fitting_attr/dfparam,model_attr/tmap,o_descriptor

Then you may want to modify the class source/train/DeepPot.py to output the descriptors by using the python interface of [model inference](https://github.com/deepmodeling/deepmd-kit/tree/devel#model-inference) 

Please let us know if it works

Han

**Comment by luigibonati** [comment]
Dear Han, 
thank you for your answer. 

I cloned the devel branch and tried to install the DeePMD-kit's python interface. I am able to install it via pip, but when I try to check the installation with 
`python -m deepmd -h `
the following error is raised
```
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/bonatl/Software/deepmd-devel/venv/lib/python3.6/site-packages/deepmd/__main__.py", line 3, in <module>
    from .train import train
  File "/home/bonatl/Software/deepmd-devel/venv/lib/python3.6/site-packages/deepmd/train.py", line 16, in <module>
    from deepmd.Trainer import NNPTrainer
  File "/home/bonatl/Software/deepmd-devel/venv/lib/python3.6/site-packages/deepmd/Trainer.py", line 33, in <module>
    import deepmd._prod_force_grad
  File "/home/bonatl/Software/deepmd-devel/venv/lib/python3.6/site-packages/deepmd/_prod_force_grad.py", line 15, in <module>
    op_grads_module = tf.load_op_library(module_file)
  File "/home/bonatl/Software/deepmd-devel/venv/lib/python3.6/site-packages/tensorflow/python/framework/load_library.py", line 56, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename)
tensorflow.python.framework.errors_impl.NotFoundError: /home/bonatl/Software/deepmd-devel/venv/lib/python3.6/site-packages/deepmd/libop_grads.so: undefined symbol: _ZN10tensorflow8internal21CheckOpMessageBuilder9NewStringB5cxx11Ev
```

Please note that this error was first occurring related to libop_api.so, then I changed the cmake part of setup.py to have -DTF_GOOGLE_BIN=true and now the error is related to libop_grads.so.

The configuration I am using is the following:
```
Ubuntu 18.04
Python 3.6.8
Tensorflow 1.8.0
gcc 7.4.0
cmake 3.10.2
```

Do you have any suggestions?

Thank you,
Luigi

**Comment by njzjz** [comment]
Hi,
Google's TensorFlow packages is built by gcc 4.8, so it is required to compile DeePMD-kit by gcc 4. (see tensorflow/tensorflow#27067 for details)
Or you can try [Anaconda's TensorFlow package](https://www.anaconda.com/tensorflow-in-anaconda/). It's built by gcc 5.4.
